flowbased-computation/flowbased-computation-impl/src/main/java/com/farao_community/farao/flowbased_computation/impl/LoopFlowComputation.java

```diff
@@ -101,26 +110,26 @@ private static void buildFlowFromNetwork(Network network, Crac crac, Map<Cnec, D
         states.forEach(state -> crac.getCnecs(state).forEach(cnec -> cnecFlowMap.put(cnec, cnec.getP(network))));
     }
 
-    public Map<String, Double> getRefNetPositionByCountry(Network network) {
+    public Map<Country, Double> getRefNetPositionByCountry(Network network) {
         //get Net Position of each country from Network
-        Map<String, Double> refNpCountry = new HashMap<>();
-        for (String country : countries) {
-            CountryAreaFactory countryAreaFactory = new CountryAreaFactory(Country.valueOf(country));
+        Map<Country, Double> refNpCountry = new HashMap<>();
+        for (Country country : countries) {
+            CountryAreaFactory countryAreaFactory = new CountryAreaFactory(country);
             double countryNetPositionValue = countryAreaFactory.create(network).getNetPosition();
             refNpCountry.put(country, countryNetPositionValue);
         }
         return refNpCountry;
     }
 
-    public Map<Cnec, Double> buildZeroBalanceFlowShift(Map<Cnec, Map<String, Double>> ptdfResults, Map<String, Double> referenceNetPositionByCountry) {
+    public Map<Cnec, Double> buildZeroBalanceFlowShift(Map<Cnec, Map<Country, Double>> ptdfResults, Map<Country, Double> referenceNetPositionByCountry) {
         Map<Cnec, Double> loopFlowShift = new HashMap<>();
-        for (Map.Entry<Cnec, Map<String, Double>> entry : ptdfResults.entrySet()) {
+        for (Map.Entry<Cnec, Map<Country, Double>> entry : ptdfResults.entrySet()) {
             Cnec cnec = entry.getKey();
-            Map<String, Double> cnecptdf = entry.getValue();
+            Map<Country, Double> cnecptdf = entry.getValue();
             double sum = 0.0;
             // calculate PTDF * NP(ref)
-            for (Map.Entry<String, Double> e : cnecptdf.entrySet()) {
-                String country = e.getKey();
+            for (Map.Entry<Country, Double> e : cnecptdf.entrySet()) {
+                Country country = e.getKey();
                 sum += cnecptdf.get(country) * referenceNetPositionByCountry.get(country);
```

Comment:
This is the reason why I added this fix : here the keys where different : 
* GLSK (cnecptdf) : key (String) was "EICode + timestamp".
* refNetPosition : key(String) was "countrycode"

Reply:
Now both keys are Country to make comparison more robust

Reply:
All good for me.